hash.js

```diff
@@ -0,0 +1,17 @@
+const crypto = require('crypto');
+
+function hash(algorithm, message, inputEncoding, outputEncoding) {
+  const func = crypto.createHash(algorithm);
+  func.update(message, inputEncoding);
+  return func.digest(outputEncoding);
+}
+
+function hashStream(algorithm, input, output) {
+  const func = crypto.createHash(algorithm);
+  input.pipe(func).pipe(output);
+}
```

Comment:
The actual behavior somebody would use is.. given a stream, get a hash. This is not the best way nor the best API to do this. Also, I would return a promise

Reply:
I'm terrible at API design, honestly. That's why I asked for your review. Will follow-up and make this one better.